src/unix/lwt_unix.mli

```diff
@@ -384,6 +384,72 @@ val write : file_descr -> Bytes.t -> int -> int -> int Lwt.t
 val write_string : file_descr -> string -> int -> int -> int Lwt.t
   (** See {!write}. *)
 
+(** Sequences of buffer slices for {!writev}. *)
+module IO_vectors :
+sig
+  type t
+  (** Mutable sequences of I/O vectors. An I/O vector describes a slice of a
+      [bytes] or [Bigarray] buffer. Each I/O vector is a triple containing a
+      reference to the buffer, an offset into the buffer where the slice begins,
+      and the length of the slice. *)
+
+  type _bigarray =
+    (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t
+  (** Type abbreviation equivalent to {!Lwt_bytes.t}. Do not use this type name
+      directly; use {!Lwt_bytes.t} instead. *)
+
+  val create : unit -> t
+  (** Creates an empty I/O vector sequence. *)
+
+  val append_bytes : t -> Bytes.t -> int -> int -> unit
+  (** [append_bytes vs buffer offset length] appends a slice of the [bytes]
+      buffer [buffer] beginning at [offset] and with length [length] to the
+      I/O vector sequence [vs]. *)
+
+  val append_bigarray : t -> _bigarray -> int -> int -> unit
+  (** [append_bigarray vs buffer offset length] appends a slice of the
+      [Bigarray] buffer [buffer] beginning at [offset] and with length [length]
+      to the I/O vector sequence [vs]. *)
+
+  val drop : t -> int -> unit
+  (** [drop vs n] adjusts the I/O vector sequence [vs] so that it no longer
+      includes its first [n] bytes. *)
+
+  val count : t -> int
+  (** [count vs] is the number of I/O vectors in the sequence [vs]. *)
```

Comment:
I suppose it also gives an easy way to know when all the data in the vectors has been written. Here is some kind of `writev` loop skeleton:

```ocaml
let rec loop () =
  Lwt_unix.writev fd vecs >>= fun n ->
  Lwt_unix.IO_vectors.drop vecs n;
  if Lwt_unix.IO_vectors.count vecs = 0 then Lwt.return_unit
  else loop ()
```

Of course, the user could keep track of how much data is in the vectors, but having `count` for this seems like less of a pain.

This could be an argument to expose a `size` or `byte_count` instead, though.